### PR TITLE
🐛 Concurrent CDK bug fixes

### PR DESCRIPTION
**Pull request by girarda**
## What
* Fix a few issues I ran into when QAing source-stripe with concurrent cdk with real connections

The fixes are:
1. Pass the Stream's transformer to the StreamFacade to cast the record values when necessary
2. Pass the Stream's namespace to the ThreadBasedConcurrentStream so its present in its AirbyteStream representation
3. Wrap the primary key in an array when setting it in the ThreadBasedConcurrentStream's AirbyteStream representation
4. Set the incremental fields to the ThreadBasedConcurrentStream's AirbyteStream representation if it has a cursor_field
5. Increase the `ThreadBasedConcurrentStream`'s default timeout to 15 minutes
6. Add `extract_state` method to AirbyteEntrypoint so we can instantiate concurrent streams only when running in full refresh

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py`
2. `airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py`
3. `airbyte-cdk/python/airbyte_cdk/entrypoint.py`

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py [hunk 1]
```diff
@@ -156,9 +158,18 @@ def get_json_schema(self) -> Mapping[str, Any]:
     def as_airbyte_stream(self) -> AirbyteStream:
         stream = AirbyteStream(name=self.name, json_schema=dict(self._json_schema), supported_sync_modes=[SyncMode.full_refresh])
 
+        if self._namespace:
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py:161]: added these for backwards compatibility when calling discover

File: airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 class ThreadBasedConcurrentStream(AbstractStream):
 
-    DEFAULT_TIMEOUT_SECONDS = 300
+    DEFAULT_TIMEOUT_SECONDS = 900
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py:28]: One of the syncs I tried failed with a 5 minutes timeout. It succeeded with 10 minutes, but increasing it to 15 so we have an additional buffer

File: airbyte-cdk/python/airbyte_cdk/entrypoint.py [hunk 3]
```diff
@@ -176,6 +176,13 @@ def set_up_secret_filter(config: TConfig, connection_specification: Mapping[str,
     def airbyte_message_to_string(airbyte_message: AirbyteMessage) -> Any:
         return airbyte_message.json(exclude_unset=True)
 
+    @classmethod
+    def extract_state(cls, args: List[str]) -> Optional[Any]:
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/entrypoint.py:180]: we can use this from the main.py to create different streams when running in full refresh or incrementally. [example for stripe](https://github.com/airbytehq/airbyte/pull/31327/files#diff-ab40a7ecb7009e3b038d40cda9e9a3dbe43725152dff44fa09a3c91bc8ea95c6R12-R16)

File: airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py [hunk 4]
```diff
@@ -280,6 +283,9 @@ class AvailabilityStrategyFacade(AvailabilityStrategy):
     def __init__(self, abstract_availability_strategy: AbstractAvailabilityStrategy):
         self._abstract_availability_strategy = abstract_availability_strategy
 
+    # The data returned by the partitions should be formatted appropriately, so we explicitly set the transformer to NoTransform
+    transformer: TypeTransformer = TypeTransformer(TransformConfig.NoTransform)
```
Comment (maxi297) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py:287]: I'm surprised to see this in `AvailabilityStrategyFacade` as I don't see it in the interface `AvailabilityStrategy`. What am I missing? Where is it used? I also do not see a test that explain how it was failing before

File: airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py (hunk 2 above)
Comment (maxi297) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py:28]: Does that mean that a partition took more than 5 minutes to be fetched? I'm not that surprised given that slices can be `[None]` but it make me wonder if this timeout shouldn't account for a whole sync time or if we need heartbeat of some sorts

**Review by maxi297 — APPROVED** [review]
I'm somewhat afraid of the timeout issue but this doesn't need to block this PR

**Pull request merged**